esmvalcore/preprocessor/_mask.py

```diff
@@ -50,22 +60,29 @@ def _get_fx_mask(fx_data, fx_option, mask_type):
     return inmask
 
 
-def _apply_fx_mask(fx_mask, var_data):
-    """Apply the fx data extracted mask on the actual processed data."""
-    # Apply mask across
-    old_mask = da.ma.getmaskarray(var_data)
-    mask = old_mask | fx_mask
-    var_data = da.ma.masked_array(var_data, mask=mask)
-    # maybe fill_value=1e+20
-
-    return var_data
+def _apply_mask(
+    mask: np.ndarray | da.Array,
+    array: np.ndarray | da.Array,
+    dim_map: Optional[Iterable[int]] = None,
+) -> np.ndarray | da.Array:
+    """Apply a (broadcasted) mask on an array."""
+    npx = get_array_module(mask, array)
+    if dim_map is not None:
```

Comment:
Why is this condition needed? The function appears to be always called with a `dim_map` argument.

Reply:
The `tests` use that function without a `dim_map`. I think it's nice to be able to call this function without broadcasting.

Reply:
If the feature is not used by any non-test code I would prefer to remove it, as it makes it more complicated to understand the code at no current benefit. If there is a use case in the future we could always add it again.

Reply:
Anyway, the code is in a much better shape with the changes in this pull request than it was before, so I'm happy to see this merged regardless of this point.